src/System.IO.Packaging/src/System/IO/Packaging/ContentType.cs

```diff
@@ -0,0 +1,744 @@
+            return false;
+        }
+
+        /// <summary>
+        /// Returns true if the input character is an ASCII digit or letter
+        /// Returns false if the input character is not an ASCII digit or letter
+        /// </summary>
+        /// <param name="character">input character</param>
+        /// <returns></returns>
+        private static bool IsAsciiLetterOrDigit(char character)
+        {
+            if (IsAsciiLetter(character))
+            {
+                return true;
+            }
+            if (character >= '0')
+            {
+                return (character <= '9');
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// Returns true if the input character is an ASCII letter
+        /// Returns false if the input character is not an ASCII letter
+        /// </summary>
+        /// <param name="character">input character</param>
+        /// <returns></returns>
+        private static bool IsAsciiLetter(char character)
+        {
+            if ((character >= 'a') && (character <= 'z'))
+            {
+                return true;
+            }
+            if (character >= 'A')
+            {
+                return (character <= 'Z');
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// Returns true if the input character is one of the Linear White Space characters - 
+        /// ' ', '\t', '\n', '\r'
+        /// Returns false if the input character is none of the above
+        /// </summary>
+        /// <param name="ch">input character</param>
+        /// <returns></returns>
+        private static bool IsLinearWhiteSpaceChar(char ch)
+        {
+            if (ch > ' ')
+            {
+                return false;
+            }
+
+            foreach (char c in s_linearWhiteSpaceChars)
+            {
+                if (ch == c)
+                    return true;
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// Lazy initialization for the ParameterDictionary
+        /// </summary>
+        private void EnsureParameterDictionary()
+        {
+            if (_parameterDictionary == null)
+            {
+                _parameterDictionary = new Dictionary<string, string>(); //initial size 0
+            }
+        }
+
+        #endregion Private Methods
+
+        //------------------------------------------------------
+        //
+        //  Private Members
+        //
+        //------------------------------------------------------
+        #region Private Members
+
+        private string _contentType = null;
+        private string _type = String.Empty;
+        private string _subType = String.Empty;
+        private string _originalString;
+        private Dictionary<string, string> _parameterDictionary = null;
+        private bool _isInitialized = false;
+
+        private const string _quote = "\"";
+        private const char _semicolonSeparator = ';';
```

Comment:
Our style guidelines say that constants should begin with a capital rather than an underscore, e.g. SemicolonSeparator.